src/actions/LoginThunkActions.ts

```diff
@@ -1,103 +1,81 @@
+import moment from 'moment';
 import { ThunkDispatch } from 'redux-thunk';
-import { setServerConfig, ServerConfig } from '../config';
+import { setServerConfig } from '../config';
 import { HTTP_CODES } from '../types/Common';
-import { KialiAppState } from '../store/Store';
+import { KialiAppState, LoginState, Session, LoginStatus } from '../store/Store';
 import { KialiAppAction } from './KialiAppAction';
 import HelpDropdownThunkActions from './HelpDropdownThunkActions';
 import GrafanaThunkActions from './GrafanaThunkActions';
 import { LoginActions } from './LoginActions';
 import * as API from '../services/Api';
 
-const performLogin = (
-  dispatch: ThunkDispatch<KialiAppState, void, KialiAppAction>,
-  username?: string,
-  password?: string
-) => {
+import * as Login from '../services/Login';
+
+type KialiDispatch = ThunkDispatch<KialiAppState, void, KialiAppAction>;
+
+const Dispatcher = new Login.LoginDispatcher();
+
+const shouldRelogin = (state?: LoginState): boolean =>
+  !(state && state.status === LoginStatus.loggedIn && new Date() < moment(state.session!.expiresOn).toDate());
+
+const loginSuccess = async (dispatch: KialiDispatch, session: Session) => {
+  const authHeader = `Bearer ${session.token}`;
+
+  try {
+    dispatch(LoginActions.loginSuccess(session));
+
+    dispatch(HelpDropdownThunkActions.refresh());
+    dispatch(GrafanaThunkActions.getInfo(authHeader));
+
+    const response = await API.getServerConfig(authHeader);
+    setServerConfig(response.data);
+  } catch (error) {
+    if (error.response && error.response.status === HTTP_CODES.UNAUTHORIZED) {
+      dispatch(LoginActions.logoutSuccess());
+    }
+  }
+};
+
+const performLogin = (dispatch: KialiDispatch, getState: () => KialiAppState, data?: any) => {
```

Comment:
Can we tighten the `data?: any` to be more specific?

Reply:
What is `data`? What kind of data? By the variable name, I have no clue because everything is data and there is no type so I still have no clue.

Reply:
Just trying to drive toward self-documenting code that is easily grokked without having to dive into the exact context and implementation.